pkg/controller/plan/kubevirt.go

```diff
@@ -1542,14 +1542,22 @@ func changeVmName(currName string, vmID string) string {
 	if nameExcludeChars.MatchString(newName) {
 		newName = nameExcludeChars.ReplaceAllString(newName, "")
 	}
-	for strings.HasPrefix(newName, "-") {
-		newName = newName[1:]
-	}
-	for strings.HasSuffix(newName, "-") {
-		newName = newName[:len(newName)-1]
-	}
+	newName = strings.Trim(newName, "-")
 	if len(newName) == 0 {
-		newName = "vm-" + vmID[:4]
+		newName = "vm-" + generateRandVmNameSuffix()
 	}
 	return newName
 }
+
+// Generates a random string of length six, consisting of lowercase letters and digits.
+func generateRandVmNameSuffix() string {
+	const charset = "abcdefghijklmnopqrstuvwxyz" + "0123456789"
+	source := rand.NewSource(time.Now().UTC().UnixNano())
+	seededRand := rand.New(source)
+
+	b := make([]byte, 6)
+	for i := range b {
+		b[i] = charset[seededRand.Intn(len(charset))]
+	}
+	return string(b)
+}
```

Comment:
as we're adding more logic to this, how about separating the name handling to a different file in this package?

Reply:
good suggestion, that will be more organized and easy to deal with that way, looking into that.

Reply:
thanks, looks better but the filenames don't comply with the convention in golang (commented elsewhere)